avocado/core/test.py

```diff
@@ -877,37 +879,50 @@ def _report(self):
             self.log.info("%s %s", self.status,
                           self.name)
 
-    def fail(self, message=None):  # pylint: disable=W0221
+    def _deprecate_params_message(func):  # pylint: disable=E0213
+        """ This decorator helps to deprecate parameter 'message' and
+        replace it with 'msg'.
+        """
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            message = kwargs.get('message', None)
+            if message is not None:
+                kwargs['msg'] = message
+                warnings.warn("The parameter 'message' has been deprecated. Please use instead 'msg'", DeprecationWarning)
+            return func(*args, **kwargs)
+        return wrapper
+
+    @_deprecate_params_message
+    def fail(self, msg=None, **kwargs):
```

Comment:
Not related to your changes, but the other decorated methods (`error`, `cancel`) are static and this one is not. We should make them consistent.